.github/workflows/ci.yml

```diff
@@ -0,0 +1,51 @@
+# For more information see: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow
+
+name: CI
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  lint-and-types:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [23.x]
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+    - run: npm ci
+    - run: npm run lint
```

Comment:
this should be done in sub steps ? I'm no sure.
@bmuenzenmeyer is more of an expert on this than I am

Reply:
its fine as long as you think linting and types go hand in hand here 👍

Reply:
Not especially, but the have the same setup and separating them into their own jobs seemed wasteful—no need to create a fresh environment for each; they don't affect each other.

Reply:
wasteful for who? actions are free.

Tests run in a matrix; linting only ever needs to run once; it’s generally best to split it for CI hygiene.

Reply:
Electricity? Spinning up a new environment may not cost us money, but it costs IRL resources—unless I'm misunderstanding what will happen?